xc/xenops_server_xen.ml

```diff
@@ -1039,6 +1042,26 @@ module VM = struct
       is_uefi;
     }
 
+  let xen_platform_of ~vm ~vmextra =
+    let open VmExtra in
+    match List.assoc_opt "device_id" vm.Xenops_interface.Vm.platformdata with
+    | Some device_id ->
+      (* The revision is always 2 if a device_id is specified *)
+      int_of_string device_id, 2
```

Comment:
The device_id field is hex AFAIK.

If the VM has a weird device_id set (e.g. "000a"), then previously this would correctly be passed as a hex value. Now it would result in a failure as int_of_string will choke on it. Furthermore if the value is something like "0010" it will become an int 10 which then results in `device-id=0x000a` when converting back to hex...

Reply:
OCaml's `int_of_string` can read hex values:

```
int_of_string "0xa" -> 10
```

So we could prefix string `device_id` with `0x` to make sure it is read as a hex value.

Reply:
You're absolutely right. I'll fix that.

Reply:
Done.